### ISSUE TITLE
[RFE] ATB/StratCon OpFor 'Strength' rating, similar to Morale

### ISSUE THREAD

**Issue by UlyssesSockdrawer** [MegaMek/mekhq#2906]
### Environment
N/A

### Description
TL;DR: Implement an OpFor 'Strength Rating' alongside the current 'Morale Rating' system, which when depleted can mark a contract for completion and prevents further scenario generation.

With both ATB and StratCon campaigns, as the player's force grows and dependent on intensity settings, 'unrealistic' enemy numbers can be generated, particularly over the course of longer contracts. Examples can be garrison contracts, with players fighting multi-regiment strength forces over the course of the contract, or vs. Clans, where forces destroyed in one month can exceed a Cluster in size.

Having a basic system for tracking the strength of the opposing force (either as an abstraction; "Extreme, High, Medium, Low, Depleted" or as a pool of available BV) would introduce more 'realism' into contracts and prevent salvage/prisoner overkill.

Having a BV pool system, perhaps matched to the total BV value of the player's TO&E with some +/-% modifier, would also be a good way to limit the maximum opposition faced. The total BV pot remaining could be visible in GM mode.

When a scenario is generated, the system would then subtract BV from the pot (e.g. a 30k BV Clan force would have its BV subtracted from a total of 150k BV 'in system/on planet'). At the end of the scenario, any 'mechs not destroyed would have their BV value returned to the pot, and any BV lost as destroyed/salvaged 'mechs would remain subtracted from the total.

I don't know how complex this would be to code, but this might go some way to bridging the gap between the current infinite hostile forces and a full 'persistent opfor' system.

### Files
Mock-up of the way this could be displayed to the player/GM:

![Enemy Strength Mock-up](https://user-images.githubusercontent.com/88290873/137596684-2b207816-9878-44aa-832a-59962d3b973a.png)

**Comment by UlyssesSockdrawer** [comment]
As an additional thought, this system could then be expanded in future with the monthly 'contract events', to give enemy forces reinforcements to increase their max BV on planet.

A similar BV pool could be used for allied forces, so protecting/evacuating allied troops from facilities in StratCon feels more meaningful as it saves allied BV for future use.

**Comment by Strycker1776** [comment]
I would like to second this idea

**Comment by IllianiBird** [comment]
This is essentially covered by the current implementation of 'morale'. Closing :)
